File: top-interview/dynamic-programing/best-time-to-buy/python/best_time.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minimum = -1
        maximum = -1

        result = 0
        for price in prices:
            if minimum == -1 or minimum > price:
                minimum = price
                maximum = -1
            else:
                if price > minimum and price > maximum:
                    maximum = price

            result = max(result, maximum - minimum)

        return result

File: top-interview/array/best-time-to-buy/go/best_time.go
package besttime

func maxProfit(prices []int) int {
	r := 0
	for i := 0; i < len(prices)-1; i++ {
		d := prices[i+1] - prices[i]

		if d > 0 {
			r += d
		}
	}

	return r
}
